server/wal/readonly_segment.go

```diff
@@ -0,0 +1,175 @@
+// Copyright 2023 StreamNative, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package wal
+
+import (
+	"fmt"
+	"io"
+	"log/slog"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/edsrzf/mmap-go"
+	"github.com/pkg/errors"
+	"go.uber.org/multierr"
+
+	"github.com/streamnative/oxia/server/wal/codec"
+)
+
+func segmentPath(basePath string, firstOffset int64) string {
+	return filepath.Join(basePath, fmt.Sprintf("%d", firstOffset))
+}
+
+func fileOffset(idx []byte, firstOffset, offset int64) uint32 {
+	return codec.ReadInt(idx, uint32((offset-firstOffset)*4))
+}
+
+type ReadOnlySegment interface {
+	io.Closer
+
+	BaseOffset() int64
+	LastOffset() int64
+
+	LastCrc() uint32
+
+	Read(offset int64) ([]byte, error)
+
+	Delete() error
+
+	OpenTimestamp() time.Time
+}
+
+type readOnlySegment struct {
+	c *segmentConfig
+	// codec      codec.Codec
```

Comment:
You can delete those comments directly. :)

Reply:
removed, PTAL